source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -1046,30 +1045,27 @@ namespace
                     const string& interfaceName = elem->getInterfaceName();
                     if (!interfaceName.empty())
                     {
+                        const RtIdentifier inputName(elem->getName());
+                        PvtInput* input = findInputOrThrow(inputName, node);
+
                         const RtIdentifier socketName(interfaceName);
                         RtOutput socket = nodegraph.getInputSocket(socketName);
                         if (!socket)
                         {
                             const RtIdentifier inputType(elem->getType());
-
-                            const uint32_t flags = elem->isA<Token>() ? RtPortFlag::UNIFORM | RtPortFlag::TOKEN : 0;
-                            RtInput input = nodegraph.createInput(socketName, inputType, flags);
-                            socket = nodegraph.getInputSocket(input.getName());
+                            RtInput graphInput = nodegraph.createInput(socketName, inputType, input->getFlags());
```

Comment:
Copy the flags from the node input. Before we were missing the case of copying the uniform flag.